### PR TITLE
Specify config files

### LINKED ISSUE #718: Shared lint config amongst projects
Hey guys,

I tried finding an answer to this question but I couldn't so apologies if this is a duplicate issue.

I've got a few different repos and I'd like them all to share the same Config file, but I'd also like to avoid duplicating the config file for maintainability sake.

My plan was:
1. Create a repo dedicated for `.sass-lint.yml`
2. Publish package to NPM
3. Install package (sass lint config)
4. Point Sass Lint to this config

Steps 1–3 have worked out fine 👌

The issue I am having is with step 4.
I've tried a few different ways, but essentially, this is what I am trying:

Script in *package.json*
```
{
  "scripts": {
    "sass-lint": "sass-lint -c ./.sass-lint.yml './src/**/*.s+(a|c)ss' --verbose --no-exit"
  }
}
```

Local `.sass-lint.yml` in root
```
options:
  config-file: ./node_modules/@private-npm-name/sass-lint-config/.sass-lint.yml
  merge-default-rules: true
```

I would like to be able to use the `config-file` option to point to an external config file and be able to extend it by setting new rules in the local `.sass-lint.yml`. I am currently doing this with eslint, the same concept. 

What's actually happening is that Sass Lint completely ignored the external config.

version of Sass Lint: 1.7.0

Any help will be much appreciated!

### PR DESCRIPTION
**Pull request by DanPurdy**
Tidies up the config loader in a non breaking way.

Allows you to specify a config file inside your main config file which you would wish to extend from.

The documentation I added explains the logic but basically it recursively merges all config files you specify into a new configuration object.

Also closes #718

Also uncovered a problem with the findFile function which will need to be investigated.

`<DCO 1.1 Signed-off-by: Dan Purdy dan@danpurdy.co.uk>`

### CODE REVIEW

**Comment by peduarte** [comment]
Has this fix been published to npm or this PR needs to be merged first?
Would really like to start using it asap 😁

**Comment by bgriffith** [comment]
Not yet! My fault! It's first on the review list...

**Comment by peduarte** [comment]
No worries!

File: lib/config-helpers.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  return {
+    options: resolvedConfig.options || {},
+    files: resolvedConfig.files || {},
+    rules: resolvedConfig.rules || {}
+  };
+};
+
+/**
+ * Checks a config file to see if another config file is specified, if it is, the file is laoded
```
Comment (bgriffith) [inline lib/config-helpers.js]: loaded

**Pull request merged**